conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -23,7 +23,8 @@ def filename(self):
         if self.arch:
             data_fname += "-{}".format(self.arch)
         data_fname += "-data.cmake"
-        return data_fname
+        # https://github.com/conan-io/conan/issues/17009
+        return data_fname.replace("|", "-")
```

Comment:
Also in ``TargetConfigurationTemplate``

Reply:
I checked and I think only the `build_type` ends up in that filename? 
https://github.com/conan-io/conan/blob/71795621a800c25021f27ab1fe73c5eefadad04e/conan/tools/cmake/cmakedeps/templates/target_configuration.py#L17
https://github.com/conan-io/conan/blob/71795621a800c25021f27ab1fe73c5eefadad04e/conan/tools/cmake/cmakedeps/cmakedeps.py#L26

Reply:
Oh, yes, you are right!